### PR TITLE
2022 GradleRIO Rewrite

### PR DESCRIPTION
**Pull request by ThadHouse**

### CODE REVIEW

**Comment by Starlight220** [comment]
I haven't looked at the codebase recently, but the last time I noticed a few instances where Strings compared with `==` in Groovy weren't migrated to `.equals()` comparison in Java. This can cause problems.

**Comment by ThadHouse** [comment]
There were a few == which I fixed.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -18,13 +18,13 @@ jobs:
...
         path: ~/.m2/
-      
+
   test_examples:
     needs: build
     runs-on: ubuntu-latest
     strategy:
      matrix:
-        language: [cpp, java, jni, kotlin]
+        language: [cpp, java] #, jni, kotlin]
```
Comment (PeterJohnson) [inline .github/workflows/ci.yml:27]: Are we permanently removing JNI and Kotlin or just temporarily?

File: examples/cpp/build.gradle [hunk 2]
```diff
@@ -1,60 +1,80 @@
...
+    //     }
+        // // Built in artifact to deploy arbitrary files to the roboRIO.
+        // fileTreeArtifact('frcStaticFileDeploy') {
+        //     // The directory below is the local directory to deploy
+        //     files = fileTree(dir: 'src/main/deploy')
+        //     // Deploy to RoboRIO target, into /home/lvuser/deploy
+        //     targets << "roborio"
+        //     directory = '/home/lvuser/deploy'
+        // }
+    //}
```
Comment (PeterJohnson) [inline examples/cpp/build.gradle:47]: Lots of commented lines here... what's the purpose?

File: examples/cpp/build.gradle [hunk 3]
```diff
@@ -1,60 +1,80 @@
...
-// Simulation configuration (e.g. environment variables).
-sim {
-    // Sets the websocket client remote host.
-    // envVar "HALSIMWS_HOST", "10.0.0.2"
-}
+// // Simulation configuration (e.g. environment variables).
+// sim {
+//     // Sets the websocket client remote host.
+//     // envVar "HALSIMWS_HOST", "10.0.0.2"
+// }
```
Comment (PeterJohnson) [inline examples/cpp/build.gradle:72]: Ditto, lots of commented lines.

File: examples/cpp/out.txt [hunk 4]
```diff
@@ -0,0 +1,4 @@
+
+> Configure project :
+Adding maven
+1st after evaluate
```
Comment (PeterJohnson) [inline examples/cpp/out.txt:4]: Should this file be committed?

**Review by PeterJohnson — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (ThadHouse) [reply .github/workflows/ci.yml:27]: Temporarily.

File: examples/cpp/build.gradle (hunk 3 above)
Comment (ThadHouse) [reply examples/cpp/build.gradle:72]: These ones I need to finish up in the templates which was going to be a this week project

File: examples/cpp/build.gradle (hunk 2 above)
Comment (ThadHouse) [reply examples/cpp/build.gradle]: Just so I could have the original templates. REmoved them.

File: examples/cpp/out.txt (hunk 4 above)
Comment (ThadHouse) [reply examples/cpp/out.txt]: No

File: src/main/java/edu/wpi/first/gradlerio/wpi/WPIPlugin.java [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+
+        project.getTasks().register("wpiVersions", task -> {
+            task.setGroup("GradleRIO");
+            task.setDescription("Print all versions of the wpi block");
+            task.doLast(new Action<Task>() {
+				@Override
+				public void execute(Task arg0) {
+                                //     wpiExtension.versions().each { String key, Tuple tup ->
+            //         println "${tup.first()}: ${tup[1]} (${key})"
+            //     }
```
Comment (PeterJohnson) [inline src/main/java/edu/wpi/first/gradlerio/wpi/WPIPlugin.java:49]: This no longer prints (which is what the command is supposed to do).

File: src/main/java/edu/wpi/first/gradlerio/wpi/WPIVersionsExtension.java [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+    private static final String wpimathVersion = "2022.0.0-alpha-2-75-gc42e053";
+    private static final String googleTestVersion = "1.9.0-5-437e100-1";
+
+    private static final String smartDashboardVersion = "2021.2.2-1-g49b44fd";
+    private static final String shuffleboardVersion = "2021.3.1-5-g675f4c8";
+    private static final String outlineViewerVersion = "2022.0.0-alpha-2-75-gc42e053";
+    private static final String robotBuilderVersion = "2021.3.1-9-g0ed948a";
+    private static final String robotBuilderOldVersion = "2021.3.1";
+    private static final String pathWeaverVersion = "2021.2.2-21-g765f50b";
+    private static final String glassVersion = "2022.0.0-alpha-2-75-gc42e053";
```
Comment (PeterJohnson) [inline src/main/java/edu/wpi/first/gradlerio/wpi/WPIVersionsExtension.java:24]: Do these versions need updating?

File: src/main/java/edu/wpi/first/gradlerio/wpi/WPIPlugin.java (hunk 5 above)
Comment (ThadHouse) [reply src/main/java/edu/wpi/first/gradlerio/wpi/WPIPlugin.java:49]: Yeah thats another thing I need to fix up. I'll do that.

File: src/main/java/edu/wpi/first/gradlerio/wpi/WPIVersionsExtension.java (hunk 6 above)
Comment (ThadHouse) [reply src/main/java/edu/wpi/first/gradlerio/wpi/WPIVersionsExtension.java:24]: Now, this is newest releases in the developmenmt repo.

**Pull request merged**